src/cmds/scripts/win_postinstall.py

```diff
@@ -0,0 +1,632 @@
+                sys.exit(6)
+            else:
+                print "Database intialization is success"
+        UpdateDSconfig()
+        print "InstallDataService completed "
+
+
+def CreateServerdbFile():
+    with open(os.devnull, 'w') as fd:
+        theproc = subprocess.Popen(
+            ["net", "stop", "pbs_server"], stdout=fd, stderr=fd)
+        theproc.communicate()
+    # Creating PBS Server database...
+    ServerPath = os.path.join(pbs_path, "exec", "sbin", "pbs_server.exe")
+    with open(os.devnull, 'w') as fd:
+        theproc = subprocess.Popen([ServerPath, "-C"], stdout=fd, stderr=fd)
+        theproc.communicate()
+        # print theproc.returncode
+        if theproc.returncode > 0:
+            print "Failed to create server database..."
+            sys.exit(13)
+        else:
+            print "PBS Server database created..."
+    with open(os.devnull, 'w') as fd:
+        theproc = subprocess.Popen(
+            ["net", "start", "pbs_server"], stdout=fd, stderr=fd)
+        theproc.communicate()
+        if theproc.returncode > 0:
+            cmd = subprocess.Popen(
+                ["sc", "query", "pbs_server"],
+                stdout=subprocess.PIPE, stderr=fd)
+            for line in cmd.stdout:
+                if "_PENDING" in line:
+                    time.sleep(30)
+                    for i in range(5):
+                        stat = subprocess.Popen(
+                            ["sc", "query", "pbs_server"],
+                            stdout=subprocess.PIPE, stderr=fd)
+                        for eachline in stat.stdout:
+                            if "_PENDING" in eachline:
+                                time.sleep(30)
+                            elif "RUNNING" in eachline:
+                                break
+                    finalcheck = subprocess.Popen(
+                        ["sc", "query", "pbs_server"],
+                        stdout=subprocess.PIPE, stderr=fd)
+                    for everyline in finalcheck.stdout:
+                        if "_PENDING" in everyline:
+                            print "Failed to start Server"
+                            sys.exit(14)
+                        elif "RUNNING" in everyline:
+                            print "Server started"
+                            break
+                elif "RUNNING" in line:
+                    print "Server started"
+                    break
+    # create server default file
+    Serverdfltfile = os.path.join(
+        pbs_path, "home", "spool", "createsvrdefaults.bat")
+    QmgrPath = os.path.join(pbs_path, "exec", "bin", "qmgr.exe")
+    if (os.path.exists(QmgrPath)):
+        pass
+    else:
+        print "Qmgr not available"
+        sys.exit(15)
+    quotedQmgrPath = "\"" + QmgrPath + "\""
+    ErrChk = "IF %ERRORLEVEL% NEQ 0 GOTO ERR\r\n"
+    with open(Serverdfltfile, 'w') as svrfd:
+        svrfd.write("@echo off\r\n")
+        content = quotedQmgrPath + " -c \"create queue workq\"\r\n"
```

Comment:
same as above: Rather than writing each line in to file can we use some kind a template then change required value in template, it will help user better readability on what is going in this dynamic script.